src/commands/force/user/create.ts

```diff
@@ -70,20 +71,25 @@ export class UserCreateCommand extends SfdxCommand {
     // because fields is type UserFields & Dictionary<string> we can access these
     const permsets: string = fields.permsets;
     const generatepassword: string = fields.generatepassword;
+    const profileName = fields.profileName;
 
     // extract the fields and then delete, createUser doesn't expect a permsets or generatepassword
     delete fields.permsets;
     delete fields.generatepassword;
     delete fields.generatePassword;
+    delete fields.profileName;
 
     try {
-      await this.user.createUser(fields);
+      this.authInfo = await this.user.createUser(fields);
+      fields.permsets = permsets;
+      fields.profileName = profileName;
+      if (fields.profileName) await this.authInfo.save({ userProfileName: fields.profileName });
```

Comment:
nit - Can this be outside of the try/catch. The catch is specific to `catchCreateUser` so it feels to me that there shouldn't be anything else in there. Doesn't really matter though.